File: nomad-samples/common/py/sync-1.1/movies/get_movie.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import requests

def get_movie(AUTH_TOKEN, TITLE):

    HEADERS = {
        "Authorization": "Bearer " +  AUTH_TOKEN,
        "Content-Type": "application/json"
    }

    BODY = {
        "filters": [
            {
                "fieldName": "contentDefinitionId",
                "operator": "Equals",
                "values": MOVIE_CONTENT_DEFINITION_ID
            },
            {
                "fieldName": "title",
                "operator": "Equals",
                "values": TITLE
            },
            {
                "fieldName": "languageId",
                "operator": "Equals",
                "values": US_ENGLISH_LANGUAGE_LOOKUP_ID
            }
        ],
        "searchResultFields": [
            {
                "name": "contentDefinitionId"
            },
            {
                "name": "contentDefinitionTitle"
            },
            {
                "name": "plot"
            },
            {
                "name": "releaseDate"
            },
            {
                "name": "genre"
            },
            {
                "name": "image",
                "searchResultFields": [
                    {
                        "name": "fullUrl"
                    }
                ]
            },
            {
                "name": "movieFile",
                "searchResultFields": [
                    {
                        "name": "fullUrl"
                    }
                ]
            }
        ]
    }

    try:
        RESPONSE = requests.post(f"{ADMIN_API_URL}/admin/search", headers= HEADERS, data= json.dumps(BODY))

        # If not found return None
        if (not RESPONSE.ok):
            raise Exception()
        
        return RESPONSE.json()
    
    except:
        api_exception_handler(RESPONSE, f"Getting Movie {TITLE} failed")

File: nomad-samples/common/js/movies/create-movie/movie/get-movie.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function getMovie(AUTH_TOKEN, ID) {
    // Check for valid parameters
    if (!AUTH_TOKEN || !ID) {
        throw new Error("Get Movie: Invalid API call");
    }

    // Create header for the request
    const headers = new Headers();
    headers.append("Content-Type", "application/json");
    headers.append("Authorization", `Bearer ${AUTH_TOKEN}`);

    // Build the payload body
    const body = {
        filters: [
            {
                fieldName: "contentDefinitionId",
                operator: "Equals",
                values: prjConstants.MOVIE_CONTENT_DEFINITION_ID
            },
            {
                fieldName: "masterId",
                operator: "Equals",
                values: ID
            },
            {
                fieldName: "languageId",
                operator: "Equals",
                values: prjConstants.US_ENGLISH_LANGUAGE_LOOKUP_ID
            }
        ],
        SearchResultFields: [
            {
                name: "contentDefinitionId"
            },
            {
                name: "contentDefinitionTitle"
            },
            {
                name: "plot"
            },
            {
                name: "releaseDate"
            },
            {
                name: "genre"
            },
            {
                name: "image",
                SearchResultFields: [
                    {
                        name: "fullUrl"
                    }
                ]
            },
            {
                name: "movieFile",
                SearchResultFields: [
                    {
                        name: "fullUrl"
                    }
                ]
            }
        ]
    };

    // Send POST request
    const response = await fetch(`${prjConstants.ADMIN_API_URL}/admin/search`, {
        method: "POST",
        headers: headers,
        body: JSON.stringify(body)
    });

    // Check for success
    if (response && response.ok) {
        // Get the response
        const movie = await response.json();

        // Return the movie
        return movie;
    }

    await apiExceptionHandler(response, `Get Movie [${ID}] failed`);
}
